mobile/src/screens/BookScreen.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Text, View, StyleSheet, Image, ScrollView } from 'react-native';
+import { useRoute } from '@react-navigation/native';
+import { MarkdownView } from 'react-native-markdown-view';
+import YoutubePlayer from 'react-native-youtube-iframe';
+import { BookScreenRouteProps } from '../navigation/HomeStackNavigator';
+import { LoadingCircle } from '../components/LoadingCircle';
+import { APIContext } from '../context/APIContext';
+import { I18nContext } from '../context/I18nContext';
+import { TextStyles } from '../styles/TextStyles';
+import { ButtonGroup } from '../components/ButtonGroup';
+import { LanguageButtons } from '../components/LanguageButtons';
+
+/**
+ * Individual book view displaying book details
+ */
+export const BookScreen: React.FC = () => {
+  // get book id and langs from route params
+  const route = useRoute<BookScreenRouteProps>();
```

Comment:
You can just add the route object to the arguments of BookScreen, no need for a hook here: https://reactnavigation.org/docs/params

```ts
export const BookScreen: React.FC = ({ route }) => {

}
```